src/RefractAPI.cc

```diff
@@ -7,17 +7,442 @@
+
+            typeAttributes->push_back(refract::IElement::Create(parameter.use == snowcrash::RequiredParameterUse ? "required" : "optional"));
+            element->attributes["typeAttributes"] = typeAttributes;
+        }
+
+        return element;
+    }
+
+    refract::IElement* ParametersToRefract(const snowcrash::Parameters& parameters)
+    {
+        refract::ObjectElement* element = new refract::ObjectElement;
+        RefractElements content;
+
+        element->element("hrefVariables");
+        std::transform(parameters.begin(), parameters.end(), std::back_inserter(content), ParameterToRefract);
+        element->renderType(refract::IElement::rFull);
+
+        element->set(content);
+
+        return element;
+    }
+
+    refract::IElement* HeaderToRefract(const snowcrash::Header& header)
+    {
+        refract::MemberElement* element = new refract::MemberElement;
+        element->set(refract::IElement::Create(header.first), refract::IElement::Create(header.second));
+
+        return element;
+    }
+
+    refract::IElement* HeadersToRefract(const snowcrash::Headers& headers)
+    {
+        refract::ObjectElement* element = new refract::ObjectElement;
+        RefractElements content;
+
+        element->element("httpHeaders");
+        std::transform(headers.begin(), headers.end(), std::back_inserter(content), HeaderToRefract);
+        element->renderType(refract::IElement::rFull);
+
+        element->set(content);
+
+        return element;
+    }
+
+    refract::IElement* AssetToRefract(const snowcrash::Asset& asset, const std::string& contentType, bool messageBody = true)
+    {
+        if (asset.empty()) {
+            return NULL;
+        }
+
+        refract::IElement* element = refract::IElement::Create(asset);
+
+        element->element("asset");
+        element->meta["classes"] = refract::ArrayElement::Create(messageBody ? "messageBody" : "messageSchema");
+
+        if (!contentType.empty()) {
+            element->attributes["contentType"] = refract::IElement::Create(contentType);
+        }
+
+        return element;
+    }
+
+    refract::IElement* PayloadToRefract(const snowcrash::Payload* payload, const snowcrash::HTTPMethod& method = "")
+    {
+        refract::ArrayElement* element = new refract::ArrayElement;
+        RefractElements content;
+
+        // Use HTTP method to recognize if request or response
+        if (method.empty()) {
+            element->element("httpResponse");
+
+            if (payload != NULL && !payload->name.empty()) {
+                element->attributes["statusCode"] = refract::IElement::Create(payload->name);
+            }
+        }
+        else {
+            element->element("httpRequest");
+            element->attributes["method"] = refract::IElement::Create(method);
+
+            if (payload != NULL) {
+                element->attributes["title"] = refract::IElement::Create(payload->name);
+            }
+
+            // FIXME: Expand href
```

Comment:
I don't think API Blueprint uses this. But I left a remark here so that I can ask the guys later.